opesci/staggeredgrid.py

```diff
@@ -915,12 +922,15 @@ def stress_loop(self):
                         + ccode(kernel.xreplace({self.t+1: self.time[1], self.t: self.time[0]})) + ';\n'
             dict1 = {'i': i, 'i0': i0, 'i1': i1, 'body': body}
             body = render(tmpl, dict1)
-            if self.ivdep and d == self.dimension-1:
+            if not self.pluto and self.ivdep and d == self.dimension-1:
                     body = '%s\n' % self.compiler._ivdep + body
-            if self.simd and d == self.dimension-1:
+            if not self.pluto and self.simd and d == self.dimension-1:
                     body = '#pragma simd\n' + body
 
-        if self.omp:
+        if self.pluto:
+            body = '#pragma scop\n'+body
+            body += '#pragma endscop'
+        elif self.omp:
```

Comment:
probably no,  even with one more little loop the generated code is huge . Also it would take much longer time if we wrap it around the entire time loop
Although I can test it later see if this will make it runs faster

Date: Fri, 11 Sep 2015 08:19:18 -0700
From: notifications@github.com
To: opesci-fd@noreply.github.com
CC: zlcghoul@hotmail.com
Subject: Re: [opesci-fd] Pluto (#45)

In opesci/staggeredgrid.py:

>                      body = '#pragma simd\n' + body
>  
> -        if self.omp:
> +        if self.pluto:
> +            body = '#pragma scop\n'+body
> +            body += '#pragma endscop'
> +        elif self.omp:

Is this why Pluto is only tiling the spatial loops? Should we add these in the template file around the entire time loop?


―
Reply to this email directly or view it on GitHub.